service/history/historyEngine.go

```diff
@@ -3045,7 +3045,9 @@ func (e *historyEngineImpl) GetDLQReplicationMessages(
 			e.logger.Error("Failed to fetch DLQ replication messages.", tag.Error(err))
 			return nil, err
 		}
-		tasks = append(tasks, task)
+		if task != nil {
+			tasks = append(tasks, task)
```

Comment:
Worth adding some log/metrics?

Reply:
it is a common case if the workflow passes the retention day. I don't we need to have log or metrics.